pymeasure/instruments/lakeshore/lakeshore_base.py

```diff
@@ -0,0 +1,117 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+import numpy as np
+from time import sleep, time
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class LakeShoreTemperatureInputChannel(Channel):
+    """ Temperature input channel on a lakeshore temperature monitor. Reads the temperature in
```

Comment:
Maybe better `LakeShoreTemperatureMeasurementChannel` or `LakeShoreTemperature(Sensing/Sensor)Channel`? For a moment I was confused that an "input" channel only has measurements (that produce outputs), not controls. :-P

Reply:
Yeah I think maybe just taking out the `Input` and `Output` from the temperature and heater channel classes is probably a good way to go. So the class names would be:

- `LakeShoreTemperatureChannel`
- `LakeShoreHeaterChannel`